packages/webamp-modern-2/src/skin/AudioPlayer.ts

```diff
@@ -222,6 +286,112 @@ export class AudioPlayer {
     return dispose;
   }
 
+  onVolumeChanged(cb: () => void): () => void {
+    const handler = () => cb();
+    this._audio.addEventListener("volumechange", handler);
+    const dispose = () => {
+      this._audio.removeEventListener("volumechange", handler);
+    };
+    return dispose;
+  }
+
+  //* this only custom listerner ================================
+
+  // execute the callback everytime the label is trigger
+  on(label:string, callback, checkPast = false): ()=>void {
+    this._listeners.has(label) || this._listeners.set(label, []);
+    this._listeners.get(label).push(callback);
+    if (checkPast)
```

Comment:
Instead of having this generic behavior (which involves leaking one set of args for each label) could we just implement this manually for the few cases where we need? Do we need this functionality for arbitrary labels?

Reply:
it was an early patch. Nowadays I knew that we has Emmit class that can we use to implement "on()" above.
I need to learn how to use Emmit to replace it.